### PR TITLE
Fix Redis cluster stop logic

### PR DESCRIPTION
**Pull request by mikezaccardo**
By catching any exceptions thrown by stopping the slaves or by stopping master, both actions are guaranteed to be attempted.

`CompoundRuntimeException ` is used because it allows us to capture and propagate both exceptions in the event that both the slaves and master fail to stop.

### CODE REVIEW

File: software/nosql/src/main/java/brooklyn/entity/nosql/redis/RedisClusterImpl.java [hunk 1]
```diff
@@ -119,8 +121,21 @@ public void stop() {
     }
 
     private void doStop() {
-        getSlaves().invoke(DynamicCluster.STOP, ImmutableMap.<String, Object>of()).getUnchecked();
-        getMaster().invoke(RedisStore.STOP, ImmutableMap.<String, Object>of()).getUnchecked();
+        MultiException multiException = new MultiException();
```
Comment (aledsage) [inline software/nosql/src/main/java/brooklyn/entity/nosql/redis/RedisClusterImpl.java]: Unfortunately can't use `org.mortbay.util` (see asfbot build error). Instead use `brooklyn.util.exceptions.CompoundRuntimeException`.

**Comment by mikezaccardo** [comment]
Fixed, thanks @aledsage

**Comment by aledsage** [comment]
Thanks; would merge but it has conflicts. 

@mikezaccardo can you resolve conflicts please? (and then ping folk on IRC to get it merged asap before you hit more conflicts, caused presumably by package renaming!)

**Comment by mikezaccardo** [comment]
Must be the renaming -- didn't have conflicts at last commit.  Will fix now.

**Comment by richardcloudsoft** [comment]
Merged. @mikezaccardo our normal process when there's a merge conflict with master is to rebase the branch rather than merge master into it - this avoids ugly `Merge remote-tracking branch 'upstream/master'` messages in the commit history. It's not a major issue here but please look out for this next time :smile:

**Comment by mikezaccardo** [comment]
Absolutely will do in the future, @richardcloudsoft. I wasn't sure of the best practice so that would have been a good question for the mailing list on my part.

**Comment by mikezaccardo** [comment]
I went to check and see if that's documented, and it's all there, black and white: https://brooklyn.incubator.apache.org/developers/how-to-contribute.html#contributing-using-github

**Pull request merged**